makefiles/pseudomodules.inc.mk

```diff
@@ -263,6 +263,7 @@ PSEUDOMODULES += test_utils_interactive_sync_shell
 
 # All auto_init modules are pseudomodules
 PSEUDOMODULES += auto_init_%
+NO_PSEUDOMODULES += auto_init_actuators_sensors
```

Comment:
Since you are making that distinction, why not split it onto

```
NO_PSEUDOMODULES += auto_init_actuators
NO_PSEUDOMODULES += auto_init_sensors
```

What about EEPROMs, displays or voltage regulators?

Reply:
> Since you are making that distinction, why not split it onto
> 
> ```
> NO_PSEUDOMODULES += auto_init_actuators
> NO_PSEUDOMODULES += auto_init_sensors
> ```

I previously did just that. But reviewer feedback was to merge those.

> What about EEPROMs, displays or voltage regulators?

:+1:, but that should IMO be follow up PRs. It was already enough pain to get a consensus on auto init for senors and actuators.

Reply:
> What about EEPROMs, displays or voltage regulators?

This is why an `auto_init_drivers` would fit better IMHO (and [I already proposed that](https://github.com/RIOT-OS/RIOT/pull/11871#issuecomment-523331732) ;))

Reply:
Having some structure is however of advantage. E.g. SAUL only needs sensors and actuators to be auto-initialized. File systems might want storage drivers to be auto-initialized but won't care about other drivers. And network drivers are already auto initalized. With only one generic group for all drivers, it will be difficult to properly model dependencies.

Many reviewers insistet that the application developer must have maximum control about what to auto initialize. If using SAUL with auto-registration of sensors and actuators would result in unrelated drivers also being auto initialized due to lack of structure, that would face some serious backlash.

Reply:
Now that the hard feature freeze is done, maybe we can push this forward again?

Being honest, I don't really care about whether this is named `auto_init_actuators`, `auto_init_sensors`, `auto_init_drivers`, or `auto_init_sensors_actuators`. I personally would prefer some fine-grained structure, e.g. `auto_init_drivers` could pull in `auto_init_sensors`, `auto_init_storage`, `auto_init_actuators` etc. which in turn would pull in the individual initialization hooks. This structure would make cherry-picking modules to initialize a bit easier.

But I think the grouping and naming is not too important now. This can easily be changed later on - and maybe with cross file arrays the initialization architecture will get a mayor cleanup anyway.

What I think is import now is some way that the individual drivers can be transformed to auto initialize them (without going through SAUL). That is IMO most of the work to do. Wiring those initialization hooks up in a different way should likely be possible with 30 minutes of work and a single PR - after all it is only one function call be module to initialize to move around.